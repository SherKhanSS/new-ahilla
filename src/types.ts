export interface Film {
  backgroundColor: string,
  cover: string,
  description: string,
  director: string,
  genre: string,
  id: number,
  isFavorite: boolean,
  poster: string,
  preview: string,
  previewImage: string,
  ratingCount: number,
  ratingScore: number,
  runTime: number,
  starring: string[],
  title: string,
  videoLink: string,
  year: number,
}

export interface Review {
  id: number,
  user: {
    id: number,
    name: string,
  },
  rating: number,
  comment: string,
  date: string,
}
