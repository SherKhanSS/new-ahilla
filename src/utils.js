export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getCurentFilm = (films, props) => {
  return films.find((movie) => movie.id === Number(props.match.params.id));
};

export const noop = () => {
  // do nothing
};
