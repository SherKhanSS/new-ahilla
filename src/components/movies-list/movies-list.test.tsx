import * as React from "react";
import * as renderer from "react-test-renderer";
import MoviesList from "./movies-list";
import {Film} from "../../types";
import {noop} from "../../utils";

const films: Film[] = [
  {
    backgroundColor: `#BDAD8F`,
    cover: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/No_Country_for_Old_Men.jpg`,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    director: `Ethan Coen`,
    genre: `Crime`,
    id: 1,
    isFavorite: false,
    poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/No_Country_for_Old_Men.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewImage: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/no-country-for-old-men.jpg`,
    ratingCount: 764976,
    ratingScore: 4.1,
    runTime: 122,
    starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    title: `The Grand Budapest Hotel`,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    year: 2007,
  },
];

const title = `The Grand Budapest Hotel`;

it(`Render MoviesList`, () => {
  const tree = renderer
    .create(<MoviesList
      films={films}
      title={title}
      onFilmCardClick={noop}
      onFilmCardMouseEnter={noop}
      onFilmCardMouseLeave={noop}
    />, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
