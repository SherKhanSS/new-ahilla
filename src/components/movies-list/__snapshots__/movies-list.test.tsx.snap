// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MoviesList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={true}
        muted={true}
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[Function]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
</div>
`;
